tools/serverpod_cli/lib/src/generator/types.dart

```diff
@@ -270,17 +270,20 @@ class TypeDefinition {
   }) {
     if (isRecordType) {
       return code_builder.RecordType((b) {
-        b.positionalFieldTypes.addAll(
-          generics
-              .where((f) => f.recordFieldName == null)
-              .map((f) => f.reference(serverCode, config: config)),
-        );
+        b.positionalFieldTypes.addAll([
+          for (final f in positionalRecordFields)
+            f.reference(
+              serverCode,
+              subDirParts: subDirParts,
```

Comment:
This is the crux of the fix right? Together with line 286?

Reply:
Yes, and also the same one in the other file.

Always tricky with such parameters with default ones (which are not set on the outermost call). Personally, I often resort to `required` even if it'd be `null` in the rare case, then at least the call site is clear.